library/common-library/templates/_agent-config.tpl

```diff
@@ -0,0 +1,88 @@
+{{/*
+This helper should return the defaults that the agent should take for all the Helm chart
+*/}}
+{{- define "common.agentConfig.defaults" -}}
+{{- /*if include "common.verboseLog" . }}
+verbose: 1
+{{- end */}}
+{{- if (include "common.nrStaging" .) }}
+staging: true
+{{- end }}
+{{- with include "common.proxy" . }}
+proxy: {{ . | quote }}
+{{- end }}
+{{/* Something with fedramp here */}}
+{{- end -}}
+
+
+
+{{/*
+This helper allow to override the defaults that this common-library adds to all the agents.
+*/}}
+{{- define "common.agentConfig.defaults.override" -}}
+{{/* **********************************************************************************
+***************************************************************************************
+THIS IS AN EXAMPLE OF HOW THE CONFIG DEFAULTS SHOULD GO.
+This should not touch main/master because it has things that could break all the charts
+that use this library.
+nri-kubernetes has been taken as an example with the variables that it has hascoded.
+***************************************************************************************
+********************************************************************************** */}}
+features: "docker_enabled:false"
+
+httpServerEnabled: true
+httpServerPort: 8003
+
+passthroughEnvironment: CLUSTER_NAME
+
+{{- if .Values.enableProcessMetrics | kindIs "bool" }}
+EnableProcessMetrics: {{ .Values.enableProcessMetrics | quote }}
+{{- end }}
+{{- end -}}
+
+
+
+{{/*
+This overrides everything that defaults and user have put in the configuration.
+
+THIS ALSO OVERRIDES USER'S PREFERENCES. USE WITH CARE.
+*/}}
+{{- define "common.agentConfig.override" -}}
+{{/* **********************************************************************************
+***************************************************************************************
+THIS IS AN EXAMPLE OF HOW THE CONFIG DEFAULTS SHOULD GO.
+This should not touch main/master because it has things that could break all the charts
+that use this library.
+nri-kubernetes has been taken as an example with the variables that it has hascoded.
+***************************************************************************************
+********************************************************************************** */}}
+overrideHostRoot: ""
+forwardOnly: true
+isSecureForwardOnly: true
+{{- end -}}
```

Comment:
I am again not quite seeing why would we want to do this. If we need to override the helper logic entirely, what I'd do is just not use the helper and render a configMap by hand.

Reply:
Duplicated of [this](https://github.com/newrelic/helm-charts/pull/757#discussion_r832153829). Resolving.